src/Discovery/src/ClientBase/DiscoveryServiceCollectionExtensions.cs

```diff
@@ -52,6 +53,7 @@ public static class DiscoveryServiceCollectionExtensions
         public static IServiceCollection AddDiscoveryClient(this IServiceCollection services, IConfiguration config, string serviceName = null, IDiscoveryLifecycle lifecycle = null)
         {
             Action<DiscoveryClientBuilder> builderAction = null;
+
             config ??= services.BuildServiceProvider().GetRequiredService<IConfiguration>();
```

Comment:
```suggestion

            // TODO: don't BuildServiceProvider() here
            config ??= services.BuildServiceProvider().GetRequiredService<IConfiguration>();
```
Should we throw an exception if config is null? Significant work may be required to avoid this call otherwise.. Or maybe it's not as much of a concern since all relevant service bindings and config should already be available? Same applies to Line 182

Reply:
I'm also thinking the config should be available as well from `ConfigureAppConfiguration` at this point